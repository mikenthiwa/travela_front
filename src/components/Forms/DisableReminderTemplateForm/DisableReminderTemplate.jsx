import React, { Fragment } from 'react';
import { PropTypes } from 'prop-types';
import Modal from '../../modal/Modal';
import SubmitArea from './FormFieldsets/SubmitArea';
import './DisableReminderTemplate.scss';


const renderHeader = (modalType) => {
  return modalType === 'disable reminder template' 
    ? 'Disable Reminder Template' 
    : 'Disable Reminder Condition';
};

const checkModalVisibility = (shouldOpen, modalType) => {
  return shouldOpen &&
        (modalType.match('disable reminder template') || modalType.match('disable reminder condtion'))
    ? 'visible' : 'invisible';
};

const renderTextArea = (conditionReason, handleInputChange) => {
  return (
    <Fragment>
      <textarea
        type="text"
        defaultValue={conditionReason}
        disabled={conditionReason ? true:false}
        className={
          conditionReason 
            ? 'delete-checklist-item__input disabled'
            :'delete-checklist-item__input'}
        onChange={handleInputChange}
      />
    </Fragment>
  );
};

const renderSubmitArea = (
  closeModal,
  disableReminderTemplate,
  disableEmailReminder,
  disableReason, 
  conditionReason,
  mode,
  modalType,
  isDisabling,
) => {
  return (
    <Fragment>
      <SubmitArea
        onCancel={closeModal}
        hasBlankFields={false}
        modalType={modalType}
        isDisabling={isDisabling}
        disableReminderTemplate={disableReminderTemplate}
        disableEmailReminder={disableEmailReminder}
        disableReason={disableReason}
        send="Disable"
        conditionReason={conditionReason}
        mode={mode}
      />
    </Fragment>
  );
};

export const FormBody = ({
  check ,
  handleInputChange, 
  mode,
  isDisabling,
  disableReason,
  disableEmailReminder,
  modalType,
  closeModal, 
  disableReminderTemplate
}) => (
  <span>
    { check ?
      <p className="delete-checklist-item__reason" />
      : <p className="delete-checklist-item__reason">Reason for Disabling</p>
    }
    { renderTextArea(check, handleInputChange) }
    <div className="delete-checklist-item__hr" />
    { renderSubmitArea(
      closeModal, 
      disableReminderTemplate, 
      disableEmailReminder, 
      disableReason, 
      check,
      mode, 
      modalType,
      isDisabling
    ) }
  </span>
);

const DisableReminderTemplateForm = ({
  closeModal, disableEmailReminder, disableReminderTemplate,
  shouldOpen, modalType, handleInputChange, isDisabling,
  disableReason, conditionReason, templateReason
}) => {
  const check = conditionReason || templateReason;
  const mode = check ? 'view' : 'add';
  return(
    <Modal
      closeModal={closeModal}
      customModalStyles={templateReason 
        ? 'delete-checklist-item__disable'
        : 'delete-checklist-item'}
      visibility={checkModalVisibility(shouldOpen,modalType)}
      title={check ? 'Disable Reason': renderHeader(disableReminderTemplate)}
    >
      <FormBody
        check={check}
        isDisabling={isDisabling}
        handleInputChange={handleInputChange}
        mode={mode}
        disableReason={disableReason}
        disableEmailReminder={disableEmailReminder}
        closeModal={closeModal}
        disableReminderTemplate={disableReminderTemplate}
        modalType={modalType}
      />
    </Modal>
  );
};

FormBody.propTypes = {
  check: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  disableEmailReminder: PropTypes.func.isRequired,
  disableReminderTemplate: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  disableReason: PropTypes.string,
  mode: PropTypes.string,
  modalType: PropTypes.string,
  isDisabling: PropTypes.bool
};

DisableReminderTemplateForm.propTypes = {
  handleInputChange: PropTypes.func.isRequired,
  disableEmailReminder: PropTypes.func,
  disableReminderTemplate: PropTypes.func,
  closeModal: PropTypes.func.isRequired,
  modalType: PropTypes.string,
  disableReason: PropTypes.string,
  conditionReason: PropTypes.string,
  templateReason: PropTypes.string,
  shouldOpen: PropTypes.bool.isRequired,
  isDisabling: PropTypes.bool,
};

FormBody.defaultProps = {
  modalType:'',
  disableReason: '',
  conditionReason:'',
  templateReason: '',
  mode: '',
  isDisabling: false
};

DisableReminderTemplateForm.defaultProps = {
  modalType:'',
  disableReason: '',
  conditionReason:'',
  templateReason: '',
  disableEmailReminder: () => {},
  disableReminderTemplate: () => {},
  isDisabling: false,
};
export default DisableReminderTemplateForm;
