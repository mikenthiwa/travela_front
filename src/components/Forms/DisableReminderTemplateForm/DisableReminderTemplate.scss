.disabled {
  background-color: #F8F8F8 !important;
  border: 3px solid #F8F8F8;
  padding: 30px 0;
  padding-left: 10px;
  margin-bottom: 1px !important;
}
