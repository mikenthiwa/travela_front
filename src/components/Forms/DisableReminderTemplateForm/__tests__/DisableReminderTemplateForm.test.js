import React from 'react';
import DisableReminderTemplateForm from '../DisableReminderTemplate';

describe('Renders',() => {
  let wrapper;

  const props = {
    closeModal: jest.fn(),
    disableEmailReminder: jest.fn(),
    shouldOpen: true, 
    modalType: 'disable reminder condtion', 
    handleInputChange: jest.fn(),
    disableReason: 'Not working', 
    conditionReason: 'Not working', 
    templateReason: 'Did not work'
  };
  
  beforeEach(() => {
    wrapper = mount( <DisableReminderTemplateForm {...props}
    />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
