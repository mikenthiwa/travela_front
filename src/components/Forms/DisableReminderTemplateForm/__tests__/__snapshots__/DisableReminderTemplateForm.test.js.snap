// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders renders correctly 1`] = `
<div
  className="overlay invisible  "
  onClick={null}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="modal invisible delete-checklist-item__disable"
    id=""
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "maxWidth": "",
      }
    }
    tabIndex="0"
  >
    <div
      className="modal-title-bar"
    >
      <div
        className="modal-title-text"
      >
        <div
          className="modal-title modal-title-word"
        >
          Disable Reason
        </div>
        <div
          className="lable-text modal-bar"
        >
          <div />
        </div>
      </div>
      <button
        className="modal-close"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="close"
          src="close.svg"
        />
      </button>
    </div>
    <div
      className="modal-content"
      id=""
    >
      <span>
        <p
          className="delete-checklist-item__reason"
        />
        <textarea
          className="delete-checklist-item__input disabled"
          defaultValue="Not working"
          disabled={true}
          onChange={[Function]}
          type="text"
        />
        <div
          className="delete-checklist-item__hr"
        />
        <fieldset>
          <div
            className="submit-area"
          >
            <button
              className="bg-btn bg-btn--inactive"
              id="cancel"
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
        </fieldset>
      </span>
    </div>
  </div>
</div>
`;
